kvdb/src/lib.rs

```diff
@@ -134,21 +115,22 @@ impl DBTransaction {
 ///
 /// The API laid out here, along with the `Sync` bound implies interior synchronization for
 /// implementation.
-pub trait KeyValueDB: Sync + Send {
+pub trait KeyValueDB<K: AsRef<[u8]>, V>: Sync + Send {
+
 	/// Helper to create a new transaction.
-	fn transaction(&self) -> DBTransaction { DBTransaction::new() }
+	fn transaction(&self) -> DBTransaction<K, V> { DBTransaction::new() }
 
 	/// Get a value by key.
-	fn get(&self, col: Option<u32>, key: &[u8]) -> io::Result<Option<DBValue>>;
+	fn get(&self, col: Option<u32>, key: &[u8]) -> io::Result<Option<V>>;
```

Comment:
Would it make sense to change this to `key: K`? Same for `get_by_prefix` and `iter_from_prefix`?

Reply:
`&K` for the `get` is probably ok (I'll try it). I am unsure for the `*_by_prefix` because this is a *partial key*.